include/xframe/xaxis_default.hpp

```diff
@@ -0,0 +1,178 @@
+/***************************************************************************
+* Copyright (c) 2017, Johan Mabille, Sylvain Corlay, Wolf Vollprecht and   *
+* Martin Renou                                                             *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XFRAME_XAXIS_DEFAULT_HPP
+#define XFRAME_XAXIS_DEFAULT_HPP
+
+#include <map>
+#include <vector>
+#include <ostream>
+
+#include "xaxis.hpp"
+
+namespace xf
+{
+    /*****************
+     * xaxis_default *
+     *****************/
+
+    template <class L, class T>
+    class xaxis_default
+    {
+    public:
+
+        static_assert(std::is_integral<T>::value, "mapped_type T must be an integral type");
+
+        using self_type = xaxis_default<L, T>;
+        using axis_type = xaxis<L, T>;
+        using label_list = std::vector<L>;
+        using size_type = typename label_list::size_type;
```

Comment:
You should also add all the aliases defined in `xaxis`, that will be required when you add the `xaxis_default` in the `xaxis_variant`.